From 04cf1a9a87f7de799c0a6920affa8d1d82b9c74b Mon Sep 17 00:00:00 2001
From: xiaohuig <xiaohui.gu@intel.com>
Date: Mon, 2 Aug 2021 15:59:42 +0800
Subject: [PATCH] Mesa: Add nongeneric internalformat for float point format

Issue: game "djsyhyjs" black screen
Cause: GL_INVALID_OPERATION in glBlitFramebuffer
       (bad src/dst multisample pixel formats)
       since internalforamt of drawRb(GL_RGBA16F) is
       different from that of readRb(GL_RGBA8)
Fix:   Add nongeneric internalformat for
       GL_RGBA16F and GL_RGBA32F which is same as
       the return value of _mesa_base_fbo_format
       according to "es_spec3.2 - chapther 16.2":
       "If the source formats are floating-point
       or normalized types, the sample values for
       each pixel are resolved in an
       implementationdependent manner. "

Change-Id: Iae58d0f7e61134f1170942f7efea7765cbbee593
Tracked-On: JIRA VVP0-3543
Signed-off-by: Xiaohui Gu <xiaohui.gu@intel.com>
---
 src/mesa/main/glformats.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/mesa/main/glformats.c b/src/mesa/main/glformats.c
index 068c01855a0..94f3f420424 100644
--- a/src/mesa/main/glformats.c
+++ b/src/mesa/main/glformats.c
@@ -1690,6 +1690,8 @@ _mesa_get_nongeneric_internalformat(GLenum format)
    /* GL 1.1 formats. */
    case 4:
    case GL_RGBA:
+   case GL_RGBA16F:
+   case GL_RGBA32F:
       return GL_RGBA8;
    case 3:
    case GL_RGB:
-- 
2.33.1

